contracts/Core/storage/VoteStorage.sol

```diff
@@ -4,12 +4,12 @@ pragma solidity ^0.8.0;
 import "../../lib/Structs.sol";
 
 contract VoteStorage {
-    //epoch -> stakerid -> commitment
-    mapping(uint256 => mapping(uint256 => bytes32)) public commitments;
-    //epoch -> stakerid -> assetid -> vote
-    mapping(uint256 => mapping(uint256 => mapping(uint256 => Structs.Vote))) public votes;
-    // epoch -> asset -> stakeWeight
-    mapping(uint256 => mapping(uint256 => uint256)) public totalInfluenceRevealed;
-    //epoch - >assetid -> voteValue -> weight
-    mapping(uint256 => mapping(uint256 => mapping(uint256 => uint256))) public voteWeights;
+    //stakerid -> commitment
+    mapping(uint32 => Structs.Commitment) public commitments;
+    //stakerid  -> vote
+    mapping(uint32 => Structs.Vote) public votes;
+    //epoch -> asset -> stakeWeight
+    mapping(uint32 => uint256) public totalInfluenceRevealed;
+    //epoch -> assetid -> voteValue -> weight
+    mapping(uint32 => mapping(uint8 => mapping(uint256 => uint256))) public voteWeights;
```

Comment:
Here as well I think we dont need to store for previous epochs
voteWeights are being used in dispute only, 
and dispute for any epoch can happen in that epoch only
https://github.com/razor-network/contracts/blob/503998d19611ef2ae6d15aa806e14fac5a8f922f/contracts/Core/BlockManager.sol#L106

Reply:
problem is the values with persist through epochs.
In dispute how can we verify this weight is for this epoch or previous epoch?

Reply:
oh we cant do so